### ISSUE TITLE
Documentation on testing pyshp

### ISSUE THREAD

**Issue by micahcochran** [GeospatialPython/pyshp#42]
This library only seems like it was designed to be tested during installation.  Since, the README.txt (used by the doctest) and the test data folder shapefiles/ are not installed by setup.py.

If you aren't in the pyshp folder, running shapefile.test() from the command line doesn't end well:
```
$ python3 -c 'import shapefile; shapefile.test()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.4/dist-packages/shapefile.py", line 1186, in test
    doctest.testfile("README.txt", verbose=1)
  File "/usr/lib/python3.4/doctest.py", line 2039, in testfile
    encoding or "utf-8")
  File "/usr/lib/python3.4/doctest.py", line 221, in _load_testfile
    file_contents = package.__loader__.get_data(filename)
  File "<frozen importlib._bootstrap>", line 1623, in get_data
FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.4/dist-packages/README.txt'
```

Perhaps the `test` function should be renamed to `_test`, to show that it is a private function.

A little more documentation (README?) on testing pyshp for the people contributing to the project.

**Comment by micahcochran** [comment]
On Python 3.4.3, doctest seems to not work very well at all.
```
1 items had failures:
  68 of 140 in README.txt
140 tests in 1 items.
72 passed and 68 failed.
***Test Failed*** 68 failures.
```

It seems that the criteria for "Expect nothing" has changed.

On Python 2.7 the testing is almost 100%.

**Comment by micahcochran** [comment]
Here are instructions to run the tests.

1. Download pyshp's zip file and uncompress. And change directory into the pyshp/ folder. 

2. Extra step for a Linux box,  convert the README.txt by running
`$ dos2unix README.txt`.  (Before conversion `$ file README.txt` will report the file as "README.txt: ASCII text, with CRLF line terminators",  after conversion it reports "README.txt: ASCII text")

3. Run tests from the pyshp/ folder.  This folder contains the README.txt and the blockgroups shapefile that are required for testing.  These files are NOT installed with pyshp.
  a. Run from prompt:  `$ python shapefile.py`
  b. Run from python console `>>> import shapefile; shapefile.test()`

I'll do a PR to add this to the README, hopefully soon.

**Comment by micahcochran** [comment]
Fixed in 1.2.5 release.
